optaweb-employee-rostering-frontend/src/store/rest/RestServiceClient.ts

```diff
@@ -61,15 +69,17 @@ export default class RestServiceClient {
     if (res.status >= 200 && res.status <= 300) {
       return Promise.resolve(res.data);
     }
-    if (this.dispatch !== null) {
-      if (typeJsonRegex.test(res.headers['content-type'])) {
+    if (this.dispatch !== null && this.store !== null) {
+      if (!this.store.getState().isConnected) {
```

Comment:
I would argue it's not REST client's business to check store state or dispatch actions. In my opinion it should simply reject the promise with as much _useful_ information as reasonably possible. Then it's operation's responsibility to act upon a rejected promise of a get/post/delete/... call and dispatch a corresponding action based on that.

Reply:
Many of these operations have a common error handler (namely: display an alert if an operation fails). Not handling it here will result in massive code duplication.